OneCommunityGlobal/HighestGoodNetworkApp
# Shengwei fix jae's account edit permission

# Description
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69882989/f94b3008-1f03-4358-8c64-3a38507f1d2b)

## Related PRS (if any):
This frontend PR is related to the #[649 ](https://github.com/OneCommunityGlobal/HGNRest/pull/649) backend PR.
To test this PR you need to checkout the backend PR.
…

## Main changes explained:
- Update the file **src/components/Badge/BadgeReport.jsx** for a warning message when a non-owner role tries to update the badge.
- Update the file **src/components/UserManagement/UserManagement.jsx** to add a warning message when a user clicks on the 'set final day' and 'pause' buttons.
- Update **src/components/UserManagement/UserTableData.jsx** to hide the reset password and delete buttons for non-Jae-related accounts on the user data table.
- Update **src/components/UserProfile/AssignBadgePopup.jsx** to include a warning message when a non-owner role tries to update the badge, disable the confirm button when submitting an update, and show a toast message after the badge update.
- Update **src/components/UserProfile/Badges.jsx** for Jae's account validation.
- Update **src/components/UserProfile/TimeEntryEditHistory.jsx** to hide the delete time entry button from non-Jae-related accounts in the user profile time entry history tag.
- Update **src/components/UserProfile/UserProfile.jsx** to optimize code performance: remove unnecessary calls to load user profile, update user profile, and load tasks APIs when a badge update is submitted.
- Update **src/components/UserProfile/UserProfile.jsx** to fix the retrieval of incorrect auth user email, which leads to an error in Jae's account validation.

## How to test:
1. check into the current branch and #[649 ](https://github.com/OneCommunityGlobal/HGNRest/pull/649) backend PR.
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log in as an Owner user for owner role testing 
5. log in as an Owner user and modify both to` permissions.js` in frontend and backend, and make it Jae related account for testing. (See step 7)
6. For the Owner role, you should 
    - go to user management -> search jae-related accounts ('jae@onecommunityglobal.org','one.community@me.com','jsabol@me.com')
    - verify reset password and delete button is hidden
    ![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69882989/05c9b949-f03c-4f29-9227-e93745a359f5)
    - click on pause and set final day button -> verify the warning message 
    - click on any user name and redirect to user profile page 
    -  verify the user information is not editable except assign badges and select featured
7. For permitted account-related Jae, you should verify 
    - modify permission.js and add your email to the list in the frontend
    - modify permission.js and add your userid in the backend (You can get your user id from view profile's url)
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69882989/529d2423-3d24-49d0-a92b-ec2ab76bb2c8)
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69882989/8fd78889-84f6-40d8-90af-552e67fb3e75)

    - go to user management -> search jae-related accounts ('jae@onecommunityglobal.org','one.community@me.com','jsabol@me.com')
    -  verify reset password and delete button is there
    - click on pause and set the final day button, and verify no warning is displayed
    - click on any user name and redirect to the user profile page 
    -  verify the user information is editable
9. For other users without edit permission, you should not be able to perform any updates. 
## Screenshots or videos of changes:

Video Owner Role:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69882989/82a033e5-c3b5-4845-93e6-f7340433d128

Video: Test with Jae-related account with Owner role:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69882989/b46a050c-7ec3-485c-8296-d130a9045369


## Note:
I will create a follow-up PR to address the edit restriction on Jae's account in the badge management component. The bug related to incorrect badge assignment is causing a block.
